### PR TITLE
Added example showing how to restore a snapshot.

### PR DESCRIPTION
**Pull request by rjschmidt**
Added another example file to show how to restore a VM to a snapshot.

### CODE REVIEW

**Comment by ifireball** [comment]
Hi @rjschmidt 

This project, like many other oVirt project is not developed directly on GitHub. Instead, development in done on [oVirt's Gerrit server](https://gerrit.ovirt.org), while GitHub is only used as a read-only mirror. 
Please kindly submit your patch there. 
If you've never used Gerrit before, [see here](https://ovirt.org/develop/dev-process/working-with-gerrit/) for an introduction.

**Comment by rjschmidt** [comment]
Alright I will, thank you.

-Ryan

File: sdk/examples/restore_snapshot.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+snapshots_service = vms_service.vm_service(vm.id).snapshots_service()
+
+# Add the new snapshot:
+snapshots_service.add(
+  types.Snapshot(
+    description='My snapshot',
+  ),
+)
+logging.info(
+    'Sent request to create snapshot \'%s\', the id is \'%s\'.',
```
Comment (mykaul) [inline sdk/examples/restore_snapshot.py]: Sent -> Send
snapshot -> a snapshot

File: sdk/examples/restore_snapshot.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    snap.description, snap.id,
+)
+
+# Poll and wait till the status of the snapshot is 'OK', which means
+# that it is completely created:
+snap_service = snaps_service.snapshot_service(snap.id)
+while snap.snapshot_status != types.SnapshotStatus.OK:
+    logging.info(
+        'Waiting till the snapshot is created, the satus is now \'%s\'.',
```
Comment (mykaul) [inline sdk/examples/restore_snapshot.py]: satus -> status

Won't this log be repeated over and over again till the snapshot is OK - is that the intention?

File: sdk/examples/restore_snapshot.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    vm_status = vm_service.get()
+logging.info('The VM has been powered off.')
+
+try:
+    # Restore the snapshot
+    snap_service.restore(
+        async=False,
+        restore_memory=False
+    )
+except:
```
Comment (mykaul) [inline sdk/examples/restore_snapshot.py]: Any specific exception you'd like to catch here?

File: sdk/examples/restore_snapshot.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    logging.info('VM could not be restored to previous snapshot.')
+
+# Poll and wait until the status of the VM is 'DOWN', meaning
+# that the snapshot has successfully been restored
+vm_status = vm_service.get()
+while vm_status.status != types.VmStatus.DOWN:
+    logging.info(
+        'Waiting for VM to be powered Down, the status is now \'%s\'.',
```
Comment (mykaul) [inline sdk/examples/restore_snapshot.py]: Powered down? It was not up - perhaps worth rephrasing?

**Pull request closed without merging**